Applications/Resnet/jni/cifar_dataloader.cpp

```diff
@@ -12,11 +12,58 @@
 
 #include "cifar_dataloader.h"
 
+#include <cstring>
 #include <nntrainer_error.h>
 #include <random>
 
 namespace nntrainer::resnet {
 
+namespace {
+
+/**
+ * @brief fill label to the given memory
+ *
+ * @param data data to fill
+ * @param length size of the data
+ * @param label label
+ */
+void fillLabel(float *data, unsigned int length, unsigned int label) {
+  if (length == 1) {
+    *data = label;
+    return;
+  }
+
+  int sum = 0;
+  /// if label is onehot
+  for (unsigned int j = 0; j < length; ++j) {
+    *data = (label == j);
+    sum += *data;
+    data++;
+  }
+
+  NNTR_THROW_IF(sum != 1, std::invalid_argument)
+    << "one hot encoding format is wrong, sum of the label is not one";
+}
+
+/**
+ * @brief fill last to the given memory
+ * @note this function increases iteration value, if last is set to true,
+ * iteration resets to 0
+ *
+ * @param[in/out] iteration current iteration
+ * @param iteration_per_epoch iteration per epoch
+ * @return bool true if iteration has finished
+ */
+bool isLast(unsigned int &iteration, unsigned int iteration_per_epoch) {
```

Comment:
how about rename to `iterate` as it seems like the main purpose is to increase iteration?

Reply:
changed to `setIteration`

Reply:
minor - `updateIteration`? `setIteration` is fine too btw.

Reply:
I'll change to updateIteration after validation :)

Reply:
cool.